test/functional/terminal/scrollback_spec.lua

```diff
@@ -403,12 +403,8 @@ describe("'scrollback' option", function()
     end
 
     curbufmeths.set_option('scrollback', 0)
-    if iswin() then
-      feed_data('for /L %I in (1,1,30) do @(echo line%I)\r')
-    else
-      feed_data('awk "BEGIN{for(n=1;n<=30;n++) print \\\"line\\\" n}"\n')
-    end
-    screen:expect{any='line30                        '}
+    feed_data(nvim_dir.."/shell-test REP 31 line\n")
```

Comment:
Should we make "REP" 1-based maybe?

Reply:
@justinmk 
What do you think?
There are not many usages of it now, so it's trivial to change, and makes it more sane in general I think.

(I've also thought about requiring/supporting a printf pattern there, where %d would be replaced with the count then)

Reply:
No objection (though personally it doesn't seem worth bothering, thus not worth the "churn"). printf seems interesting, but then I also wonder if it's worth the extra minutes/hours spent debugging potential bugs :)

Reply:
Agreed with regard to churn, but we typically have to add/substract 1 currently, and it will become more tests using it in the future (i.e. more churn when changing it later only).

A printf would have allowed to not change the expectations in these tests (i.e. less churn there).
For simplicity it should require a pattern probably always, and would not use/replace the counter if there is no `%d` then.  For future extension of it it might be useful to explicitly replace `{lnum}`, but that is likely not trivial in C?!